backend/src/main/java/com/huellapositiva/application/controller/ESALApiController.java

```diff
@@ -83,6 +93,52 @@ public void registerESAL(@RequestBody ESALRequestDto dto,
         }
     }
 
+    @Operation(
+            summary = "Upload ESAL logo",
+            description = ". Roles allowed CONTACT_PERSON and CONTACT_PERSON_NOT_CONFIRMED.",
+            tags = "ESAL",
+            parameters = {
+                    @Parameter(name = "X-XSRF-TOKEN", in = ParameterIn.HEADER, required = true, example = "a6f5086d-af6b-464f-988b-7a604e46062b", description = "For take this value, open your inspector code on your browser, and take the value of the cookie with the name 'XSRF-TOKEN'. Example: a6f5086d-af6b-464f-988b-7a604e46062b"),
+                    @Parameter(name = "XSRF-TOKEN", in = ParameterIn.COOKIE,required = true, example = "a6f5086d-af6b-464f-988b-7a604e46062b", description = "Same value of X-XSRF-TOKEN")
+            },
+            security = {
+                    @SecurityRequirement(name = "accessToken")
+            }
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "204",
+                            description = "No content, logo uploaded successfully."
+                    ),
+                    @ApiResponse(
+                            responseCode = "400",
+                            description = "Bad request, logo is not valid"
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Internal server error, could not upload the logo.",
+                            content = @Content(mediaType = "application/json")
+                    )
+            }
+    )
+    @PostMapping(path = "/logo", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    @RolesAllowed({"CONTACT_PERSON", "CONTACT_PERSON_NOT_CONFIRMED"})
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void uploadLogo(@RequestPart("logo")MultipartFile logo,
+                           @Parameter(hidden = true) @AuthenticationPrincipal String accountId) throws IOException {
+        try {
+            uploadLogoAction.execute(logo, accountId);
+        } catch (InvalidFieldException ex) {
+            log.error(ex.getMessage());
```

Comment:
Log the error when it occurs

Reply:
Should the log be in the UploadLogoAction method then? And throw a ResponseStatusException from there. This error also happens in VolunteerApiController, in the uploadPhoto action.

Reply:
I did not check where the error came from, just saw that you caught it and logged it. You should log it in the place that occurs which is the place where you threw `new InvalidFieldException`

Reply:
Solved